### PR TITLE
Adding width property to linear & circular progress

### PR DESCRIPTION
**Pull request by namnguyen20999**
```
from taipy.gui import Gui

value = 52
title = "Progress"
title_anchor = "bottom"  # Options: "left", "right", "top", "bottom", "none"
linear = True
show_value = True

page = """
<|progress|value={value}|linear={linear}|show_value|title={title}|color=green|width=100|>
"""

if __name__ == '__main__':
    Gui(page).run(port=3000, debug=True, use_reloader=True)
```

image

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/Progress.tsx [hunk 1]
```diff
@@ -110,7 +118,7 @@ const Progress = (props: ProgressBarProps) => {
 
     return showValue && value !== undefined ? (
         linear ? (
-            <Box sx={boxWithFlexDirectionSx}>
+            <Box sx={boxWithFlexDirectionSx} data-testid="linear-progress-container">
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Progress.tsx]: Do we really need this attribute??

File: frontend/taipy-gui/src/components/Taipy/Progress.tsx [hunk 2]
```diff
@@ -143,7 +156,7 @@ const Progress = (props: ProgressBarProps) => {
             </Box>
         )
     ) : linear ? (
-        <Box sx={boxWithFlexDirectionSx}>
+        <Box sx={boxWithFlexDirectionSx} data-testid="linear-progress-container">
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Progress.tsx]: Same comment

File: taipy/gui/viselements.json [hunk 3]
```diff
@@ -1374,6 +1374,12 @@
                         "type": "dynamic(bool)",
                         "default_value": "True",
                         "doc": "If False, this progress indicator is hidden from the page."
+                    },
+                    {
+                        "name": "width",
+                        "type": "Union[str,int]",
+                        "default_value": "None",
+                        "doc": "The width of the progress indicator, in CSS units."
```
Comment (FabienLelaquais) [inline taipy/gui/viselements.json]:
```suggestion
                        "doc": "The width of this progress indicator, in CSS units."
```

for homogeneity with the other similar props.

**Review by FabienLelaquais — DISMISSED** [review]
Looking good.
Minor comments... let me know if I should re-review

File: frontend/taipy-gui/src/components/Taipy/Progress.tsx (hunk 1 above)
Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Progress.tsx]: This is used only for unit tests only, making it more stable than the auto-generated className from React, which might change due to styling or refactoring

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Progress.tsx]: Sure. But then.... all application get this... I mean users
Can you not add the class name when generating the component when testing (yes you can)?

Comment (namnguyen20999) [reply frontend/taipy-gui/src/components/Taipy/Progress.tsx]: Updated, I've removed it from the component

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**